File: axhttp/android/okhttp/java-okhttp/demo/java-okhttp-demo/src/main/java/com/axsecurity/sdk/axhttp/okhttp/demo/MainActivity.java
package com.axsecurity.sdk.axhttp.okhttp.demo;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.axsecurity.sdk.axhttp.okhttp.AXHTTPService;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class MainActivity extends AppCompatActivity {

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn = findViewById(R.id.run_test_button);
        btn.setOnClickListener((View view) -> {

            OkHttpClient.Builder OkHttpClientBuilder = new OkHttpClient.Builder();
            OkHttpClientBuilder.readTimeout(60, TimeUnit.SECONDS);
            OkHttpClientBuilder.writeTimeout(60, TimeUnit.SECONDS);

            // *** COMMENT THE LINE BELOW FOR AgentSDK ***
            OkHttpClient client = OkHttpClientBuilder.build();

            // *** UNCOMMENT THE LINE BELOW FOR AgentSDK ***
            // AXHTTPService.setOkHttpClientBuilder(OkHttpClientBuilder);
            // OkHttpClient client = AXHTTPService.getOkHttpClient();

            // make a new Request
            Request request = new Request.Builder()
                    .url("https://example.com")
                    .build();

             client.newCall(request).enqueue(new Callback() {
                 @Override
                 public void onResponse(@NonNull Call call, @NonNull Response response) throws IOException {

                     runOnUiThread(()->{
                         TextView textView = findViewById(R.id.textView);
                         textView.setText("Response code = " + response.code());
                     });
                 }
                 @Override
                 public void onFailure(@NonNull Call call, @NonNull IOException e) {
                     runOnUiThread(()->{
                         TextView textView = findViewById(R.id.textView);
                         textView.setText("Response error = " + e.toString());
                     });
                 }
             });
        });
    }

}

File: axhttp/android/okhttp/kotlin-okhttp/demo/kotlin-okhttp-demo/src/main/java/com/axsecurity/sdk/axhttp/okhttp/demo/MainActivity.kt
package com.axsecurity.sdk.axhttp.okhttp.demo
import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.activity.ComponentActivity
import com.axsecurity.sdk.axhttp.okhttp.AXHTTPService
import okhttp3.Call
import okhttp3.Callback
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response
import java.io.IOException
import java.util.concurrent.TimeUnit


class MainActivity : ComponentActivity() {
    // Using your request url here
    private val  url: String = "https://example.com"
    private lateinit var textView: TextView
    private lateinit var apiButton: Button
    private lateinit var activity: Activity

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activity = this
        setContentView(R.layout.activity_main)
        apiButton = findViewById(R.id.run_test_button)
        textView = findViewById(R.id.textView)

        apiButton.setOnClickListener {
            onRequestApi()
        }
    }

    fun onRequestApi() {
        // *** COMMENT THE LINE BELOW FOR SDK ***
        val client = OkHttpClient.Builder().build()

        // *** UNCOMMENT THE LINE BELOW FOR SDK ***
        // val okBuilder = OkHttpClient.Builder()
        // okBuilder.callTimeout(30, TimeUnit.SECONDS)
        // AXHTTPService.setOkHttpClientBuilder(okBuilder)
        // val client = AXHTTPService.getOkHttpClient()


        val builder = Request.Builder().url(url).build()

        client.newCall(builder).enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) {
                val msg = "Http response failed: " + e.message
                Log.d(TAG, msg)
                activity.runOnUiThread {
                    textView.setText(msg)
                }
            }
            @Throws(IOException::class)
            override fun onResponse(call: Call, response: Response) {
                val msg = "Http status code " + response.code
                Log.d(TAG, msg)
                activity.runOnUiThread {
                    textView.setText( msg)
                }
            }
        })
    }

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }
}
